### PR TITLE
test/cp: advanced xattr tests with a simple in memory fs using fuse

### PR DESCRIPTION
**Pull request by sssemil**
Continuation of what was started here https://github.com/uutils/coreutils/pull/4095 and then moved here https://github.com/uutils/coreutils/pull/4099.

Basically, this fuse fs implementation aims at simulating various xattr test cases for cp.

### CODE REVIEW

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -475,6 +475,12 @@ jobs:
         ## Install `rust` toolchain
         rustup toolchain install nightly --profile minimal
         rustup default nightly
+    - name: Install/setup prerequisites
+      shell: bash
+      run: |
+        case '${{ matrix.job.os }}' in
+          ubuntu-latest) sudo apt-get -y update ; sudo apt-get -y install fuse libfuse-dev pkg-config ;; # needed for fuser
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:482]: can you be more descriptive in your comment?
(why do we use fuser?)

Comment (sssemil) [reply .github/workflows/CICD.yml:494]: yes, I will cleanup the comments after I get the CI to work

**Comment by sssemil** [comment]
@sylvestre or @tertsdiepraam maybe one of you could help me here, I'm trying to install fuse and libfuse-dev dependencies, but they are being completely ignored in these architecture specific builds. Is there some special way of installing dependencies for them? I feel like this should be pretty straightforward, but I don't see it.

**Comment by tertsdiepraam** [comment]
Those tests are run with [`cross`](https://github.com/cross-rs/cross), because they are not supported by GH actions, so you have to install them in the docker image that the tests are running in somehow.

**Review by sylvestre — CHANGES_REQUESTED** [review]
still draft

**Comment by sssemil** [comment]
Yep, just a bit busy lately

**Comment by sylvestre** [comment]
@sssemil ping ?

**Comment by sssemil** [comment]
> @sssemil ping ?

hey @sylvestre feel free to take over if you have time, I'm still unavailable 😔

**Comment by sylvestre** [comment]
inactive, so, closing

**Pull request closed without merging**